content/blog/infrastructure-testing-got-better/index.md

```diff
@@ -0,0 +1,121 @@
+1. [AWS Lambda](https://aws.amazon.com/lambda/) Step Functions for workflows; and
+1. [AWS Lake Formation](https://aws.amazon.com/lake-formation/) for creating a secure data catalog available for analytics tools.
+
+With this shift, it makes sense to throw out old configurations and pick the best tools for the job rather than be encumbered by legacy DSLs. We use Pulumi as our Infrastructure as Code tool of choice because it has all of the capabilities we need:
+
+* Pulumi supports TypeScript - the modern programming language that the team prefers for its static typing capabilities.
+* It works ‘out of the box’ with modern Integrated Development Environments (IDEs) like Visual Studio Code.
+* Because Pulumi uses real programming languages, we can unit test our code
+* By virtue of working with TypeScript, it natively supports Mocha - our testing framework of choice.
+
+## Accelerating Test and Verification
+
+There are many benefits to using Pulumi to develop and test infrastructure. First, the team has the immediate benefit of modern language and IDE support: things like missing arguments are now visible in-line as we define resources.  In addition, state tracking and version control capabilities, for all cloud provider resources, make for a terrific experience in terms of productivity.
+
+Second, creating standard infrastructure stacks for the team helps everyone move faster.  We create environments for each developer in separate AWS accounts so they can develop features without impacting other developers and we don’t have to wait for central IT to provision resources. Pulumi’s support for ephemeral environments means that developers can spin-up and spin-down stacks quickly, in order to verify that changes to the desired state are actually reflected in the provisioned resources.  This makes integration testing far easier, and ephemeral environments keep cloud costs down because they are easily destroyed when they are no longer needed.
+
+Ideally, a unit test should demonstrate that a new component or resource works in isolation from other dependencies.  Resource mocking mimics complex infrastructure in lieu of spinning up the dedicated infrastructure for a test (which would be slow and potentially expensive.) This step is critical for preventing configuration mistakes from reaching production.
+
+The unit test suite for our data lake consists of over 700 tests and previously took almost 20 minutes to execute.  With improvements to StackReference mocking and the new ‘monitor mock’ feature, these same tests now run in about 20 seconds.  _That’s a 6000% improvement over our previous performance!_
+
+## Getting Started with Resource Mocking
+
+Continuously testing resource definitions is a best practice and it’s incredibly easy to get started. In this example, we mock a resource name to attach to an IAM Policy Role. The test returns the ARN and the name of any resource, which is the Policy Role.
+
+test-setup.ts
```

Comment:
I was under the impression that we were going to add a different test example here. @spara what is the plan here?